src/parameterized_tendencies/microphysics/microphysics_wrappers.jl

```diff
@@ -261,6 +263,44 @@ function compute_precipitation_sources!(
     #! format: on
 end
 
+"""
+    compute_precipitation_heating(Seₜᵖ, ᶜwᵣ, ᶜwₛ, ᶜu, qᵣ, qₛ, ᶜts, thp)
+
+ - Seₜᵖ - cached storage for precipitation energy source terms
+ - ᶜwᵣ, ᶜwₛ - rain and snow terminal velocities
+ - ᶜu - air velocity
+ - qᵣ, qₛ - precipitation (rain and snow) specific humidities
+ - ᶜts - thermodynamic state (see td package for details)
+ - ᶜ∇T - cached temporary variable to store temperature gradient
+ - thp - structs with thermodynamic and microphysics parameters
+
+ Augments the energy source terms with heat exchange between air
+ and precipitating species, following eq. 36 from Raymond 2013
+ doi:10.1002/jame.20050 and assuming that precipitation has the same
+ temperature as the surrounding air.
+"""
+function compute_precipitation_heating!(
+    ᶜSeₜᵖ,
+    ᶜwᵣ,
+    ᶜwₛ,
+    ᶜu,
+    ᶜqᵣ,
+    ᶜqₛ,
+    ᶜts,
+    ᶜ∇T,
+    thp,
+)
+    # TODO - at some point we want to switch to assuming that precipitation
+    # is at wet bulb temperature
+
+    # compute full temperature gradient
+    @. ᶜ∇T = CT123(ᶜgradᵥ(ᶠinterp(Tₐ(thp, ᶜts))))
+    @. ᶜ∇T += CT123(gradₕ(Tₐ(thp, ᶜts)))
+    # dot product with effective velocity of precipitation
+    # (times q and specific heat)
+    @. ᶜSeₜᵖ -= dot(ᶜ∇T, (ᶜu - C123(Geometry.WVector(ᶜwᵣ)))) * cᵥₗ(thp) * ᶜqᵣ
```

Comment:
The docs say using c_p here. Do we know which one we should use?

Reply:
`p` in `c_p` there was standing for precipitating species not constant pressure. Sorry thats confusing. I thought we said it should be under constant volume in all the places other than radiative tendnecies?

Reply:
Ah ok. Well, I'm not sure if I will ever understand when to use which...but that sounds good to me:)